docs/source/howto/codes.rst

```diff
@@ -45,8 +45,92 @@ Design guidelines
 Running external codes
 ======================
 
-`#3987`_
+The process of running an external code through AiiDA involves loading and instantiating the appropriate ``CalcJob`` wrapping class for the code, filling the inputs necessary, and then running or submitting the constructed object.
 
+If the ``CalcJob`` has been registered in AiiDA (see section :ref:`on how to register plugins<how-to:plugins:entrypoints>`) you can use the ``CalculationFactory``:
+
+.. code-block:: python
+
+    from aiida.plugins import CalculationFactory
+    calculation_builder = CalculationFactory('arithmetic.add').get_builder()
+
+Otherwise, you can just load the class and get the builder directly:
+
+.. code-block:: python
+
+    from aiida.calculations.plugins.arithmetic.add import ArithmeticAddCalculation
+    calculation_builder = ArithmeticAddCalculation.get_builder()
+
+Then you need to start filling out the builder with all the inputs that you need to specify to run your calculation.
+In the case of the ``ArithmeticAddCalculation``, there are 4 required inputs:
+
+.. code-block:: python
+    :linenos:
+
+    # Specific for ArithmeticAdd
+    spec.input('x', valid_type=(orm.Int, orm.Float), help='The left operand.')
+    spec.input('y', valid_type=(orm.Int, orm.Float), help='The right operand.')
+
+    # General for all CalcJobs
+    spec.input('code', valid_type=orm.Code, help='The `Code` to use for this job.')
+    spec.input('metadata.options.resources', valid_type=dict, required=True, validator=validate_resources,
+        help='Set the dictionary of resources to be used by the scheduler plugin, like the number of nodes, '
+             'cpus etc. This dictionary is scheduler-plugin dependent. Look at the documentation of the '
+             'scheduler for more details.')
```

Comment:
Instead of this I would suggest to use `verdi plugin list aiida.calculations arithmetic.add`
This way we don't have to worry about updating this when it changes, and it will also apply to any other calculation plugin

Reply:
Would also remove the mention of resources, which will be simplified by Marnik's changes that will soon be merged in `develop`.

Reply:
Ok, so, the drawback of removing the `resources` is that in this example only nodes would be provided, which might make people think you can only pass nodes to the calculations. Are we ok with this? Should I maybe add a change to one of the optional parameters (like the `output_filename`)?

Reply:
I think that is actually ok for this introductory how-to. We could make another on in the future, something like "How to change scheduler specific inputs", which will then provide a natural way to introduce `metadata` inputs.